### PR TITLE
The Next Episode

### PR DESCRIPTION
**Pull request by jaredpalmer**
Pretty much a full rewrite...but soo many goodies to share with everyone!

- [x] **Universal Hotness** (hot reload both server and client during development)
- [x] React Router 4, with @palmerhq's HOC
- [x] Jest
- [x] Fiber
- [x] import() support
- [x] React Helmet
- [x] React SVG Loader
- [ ] Glamor
- [ ] Rapscallion?
- [ ] Express Security (helmet, hpp, csp)
- [ ] babel-plugin-tranform-define for universal config
- [x] File loader, url loader
- [ ] Production builds
- [ ] Docker
- [ ] Throng
- [ ] Docs
- [ ] Document route registration regardless of ssr
- [ ] Recursively resolve promises on nested routes without context shenanigans **Bonus** 
--------
- [ ] frameworky stuff...either merge with backpack or become backpack's first plugin.

### CODE REVIEW

**Comment by justingreenberg** [comment]
keen to review this whenever you're ready @jaredpalmer :)

have you considered implementing your rapscallion solution with sunil's css stream?

**Comment by jaredpalmer** [comment]
@justingreenberg yes

**Comment by jaredpalmer** [comment]
@justingreenberg going to probably tryout glamor-inline tonight

**Comment by jaredpalmer** [comment]
cc @ericclemmons 

- Am I doing this HMR correctly?
- How do I manipulate look and feel of`[HMR] updating...` console output to improve clarity (on server mostly)?

File: server/server.js [hunk 1]
```diff
@@ -1,33 +1,95 @@
...
+// Setup server side routing.
+server.get('*', (req, res) => {
+  const context = {};
+  const matches = routes.map(
+    // we'd probably want some recursion here so our routes could have
+    // child routes like `{ path, component, routes: [ { route, route } ] }`
+    // and then reduce to the entire branch of matched routes, but for
+    // illustrative purposes, sticking to a flat route config
+    (route, index) => {
+      const match = matchPath(req.url, route.path, route);
```
Comment (lennu) [inline server/server.js]: Hey, I've been following this branch and I think there is a problem on this line:

The syntax for matchPath is incorrect:
https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/matchPath.md

There is no third argument and the second argument should be an object which has path property?

This works fine:
const match = matchPath(req.url, route);

**Comment by jaredpalmer** [comment]
This has been moved to https://github.com/jaredpalmer/razzle. 

I am going to be updating RPS as a non-trivial example of razzle in action.

**Pull request closed without merging**